source/interfaces/Method.ts

```diff
@@ -1,9 +1,10 @@
 import MethodOverride from "./MethodOverride";
 import MethodProperties from "./MethodProperties";
+import { MethodType } from "../enums"
 import Search from "./Search";
 
 export default interface Method extends MethodProperties {
-  type: "get" | "post" | "put" | "delete" | "patch",
+  type: MethodType | "get" | "post" | "put" | "delete" | "patch",
```

Comment:
This way we can use a `string` or `MethodType` enum:
```ts
export default interface Method extends MethodProperties {
  type: MethodType | "get" | "post" | "put" | "delete" | "patch",
  search: Search,
  overrides?: Array<MethodOverride>,
}

const method1: Method = {
  type: 'get'
}

const method2: Method = {
  type: MethodType.GET
}
```

If I just accept `MethodType` I can not use the `string` way:
```ts
export default interface Method extends MethodProperties {
  type: MethodType,
  search: Search,
  overrides?: Array<MethodOverride>,
}

const method1: Method = {
  type: 'get' // error
}

const method2: Method = {
  type: MethodType.GET
}
```